crates/services/gas_price_service/src/v1/uninitialized_task/fuel_storage_unrecorded_blocks.rs

```diff
@@ -11,15 +11,38 @@ use fuel_core_storage::{
         Modifiable,
         WriteTransaction,
     },
+    Error as StorageError,
     StorageAsMut,
     StorageAsRef,
+    StorageMutate,
 };
 use fuel_core_types::fuel_merkle::storage::StorageMutateInfallible;
 use fuel_gas_price_algorithm::{
     v1,
     v1::UnrecordedBlocks,
 };
 
+pub trait AsUnrecordedBlocks {
+    type Wrapper<'a>: UnrecordedBlocks
+    where
+        Self: 'a;
+
+    fn as_unrecorded_blocks(&mut self) -> Self::Wrapper<'_>;
+}
+
+impl<S> AsUnrecordedBlocks for S
+where
+    S: StorageMutate<UnrecordedBlocksTable, Error = StorageError>,
+{
+    type Wrapper<'a> = FuelStorageUnrecordedBlocks<&'a mut Self>
+        where
+            Self: 'a;
+
+    fn as_unrecorded_blocks(&mut self) -> Self::Wrapper<'_> {
+        FuelStorageUnrecordedBlocks::new(self)
+    }
+}
+
```

Comment:
What does this trait give us that we didn't get with the newtype wrapper? I feel like the newtype pattern is the idiomatic way to deal with the orphan rule problem in Rust, but maybe this solution freed up some constraints that I can't see.

Reply:
Without this helpers trait, you need to mock `StorageMutate<UnrecordedBlocksTable, Error = StorageError>` in tests, instead of just mocking `AsUnrecordedBlocks`. You can remove it and see the problem=)